### ISSUE TITLE
NTP - Time Server not working

### ISSUE THREAD

**Issue by iteqla** [ccrisan/motioneyeos#2819]
### Preliminary Docs

I confirm that I have read the [CONTRIBUTING](https://github.com/ccrisan/motioneyeos/blob/master/.github/CONTRIBUTING.md) guide before opening this issue.

I confirm that I have read the [FAQ](https://github.com/ccrisan/motioneyeos/wiki/FAQ) before opening this issue.

### motionEyeOS Version

I am running motionEyeOS version: motionEyeOS dev20201026 - motionEye Version 0.42.1 - Motion Version 4.3.1

### Board Model

I am using the following board/model: Raspberry PI 3B+

### Camera

I am using the following type of camera: MMAL

My camera model is: Raspberry Pi Camera V2

### Network Connection

My motionEyeOS unit is connected to the network via: WiFi

### Peripherals

I am not using any peripherals 

### Log Files

I consider the following log files relevant to this issue:

 * (attach e.g. motioneye.log)
 * (attach e.g. boot.log)
 
[boot.log](https://github.com/ccrisan/motioneyeos/files/7355165/boot.log)
[motioneye.log](https://github.com/ccrisan/motioneyeos/files/7355166/motioneye.log)

Hi,
The time is not synchronizing with the NTP server and the date is set to 1970. I tried different combinations of NTP/SNTP on different servers (pool.ntp.org, 0.pool.ntp.org, uk.pool.ntp.org, time.cloudfare.com). It doesn't seem to make any difference, whatever the settings. 
I am using pihole as DNS and all the named time servers are whitelisted and resolved correctly.
The Raspberry Pi where MotionEyeOS is installed has a fixed IP 192.168.1.30.

I read the previous threads on this topic and tried the solutions mentioned in those old posts but to no avail:
When trying to restart the daemon via SSH it seems like there is none at all

```
[root@pi-bianco ~]# ntpdate
-sh: ntpdate: command not found
[root@pi-bianco ~]# ntp
-sh: ntp: command not found
[root@pi-bianco ~]# ntpd
-sh: ntpd: command not found
[root@pi-bianco ~]# service ntp restart
No such service: ntp

```
The fact that I get commend not found all the times gives me a headache.
Thanks in advance.

**Comment by starbasessd** [comment]
NTP is not a service used by motionEyeOS. 
If you 
ping pool.ntp.org 
what do you get?
If you do 
nslookup pool.ntp.org
what do you get?
I get for ping:
 ping pool.ntp.org
PING pool.ntp.org (159.203.82.102): 56 data bytes
64 bytes from 159.203.82.102: seq=0 ttl=48 time=75.288 ms
64 bytes from 159.203.82.102: seq=1 ttl=48 time=130.451 ms
64 bytes from 159.203.82.102: seq=2 ttl=48 time=95.603 ms
^C
--- pool.ntp.org ping statistics ---
3 packets transmitted, 3 packets received, 0% packet loss
round-trip min/avg/max = 75.288/100.447/130.451 ms
for nslookup I get:
nslookup pool.ntp.org
Server:         192.168.2.103
Address:        192.168.2.103:53

Non-authoritative answer:

Non-authoritative answer:
Name:   pool.ntp.org
Address: 38.229.56.9
Name:   pool.ntp.org
Address: 157.245.125.229
Name:   pool.ntp.org
Address: 23.131.160.7
Name:   pool.ntp.org
Address: 45.15.168.198

I use pi-hole, also. I don't have to whitelist any of them.

**Comment by starbasessd** [comment]
I also tried HTTP (defaults to google.com) works without whitelisting, and I have a LOT of google stuff (googleapis.com for one) blacklisted. 
I am using dev20201026, MMAL Camera, wifi only, and 6 cameras it talks to. (Not the highest framerate or resolution, though). It sets time & date on boot with no issues.

**Comment by starbasessd** [comment]
BTW, if you are into programming, the program that sets time & date is /etc/init.d/S50date which runs during bootup.

**Comment by iteqla** [comment]
The whitelisting serves only to make sure that it's not pihole blocking the NTP. That's all.

Ping returns the following
```
[root@pi-bianco ~]# ping pool.ntp.org
PING pool.ntp.org (185.103.119.60): 56 data bytes
```

but then it freezes and has to be interrupted to do anything else...
and this is what I get with traceroute
```
[root@pi-bianco ~]# nslookup pool.ntp.org
Server:         192.168.1.2
Address:        192.168.1.2:53

Non-authoritative answer:
Name:   pool.ntp.org
Address: 193.47.147.20
Name:   pool.ntp.org
Address: 185.53.93.157
Name:   pool.ntp.org
Address: 85.199.214.101
Name:   pool.ntp.org
Address: 193.150.34.2

Non-authoritative answer:

[root@pi-bianco ~]#
```

**Comment by starbasessd** [comment]
What does 
route -n 
show for gateway?

**Comment by iteqla** [comment]
```
[root@pi-bianco ~]# route -n
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
0.0.0.0         192.168.1.1     0.0.0.0         UG    0      0        0 wlan0
192.168.1.0     0.0.0.0         255.255.255.0   U     0      0        0 wlan0
```

**Comment by starbasessd** [comment]
All the settings seem right, assuming you are using DHCP to get the IP address and everything else seems to be working. You can get to the WebGUI and add cameras?
I don't know why ping hangs for you. 
Does it hang by IP address, too? 
ping 193.47.147.20
or 
38.229.56.9
or
ping google.com

**Comment by iteqla** [comment]
The ping to external addresses hangs, while if I ping internal ones it has no problem. This happens only from the MotionEyeOS machine, as with any other device I can ping everything everywhere...

I was giving the IP address manually (192.168.1.30). I changed the setting to let the DHCP assign an IP and set SNTP to pool.ntp.org... now it works and the time is showing correctly.

...and now I can ping again!
```
[root@pi-bianco ~]# ping pool.ntp.org
PING pool.ntp.org (188.114.116.1): 56 data bytes
64 bytes from 188.114.116.1: seq=0 ttl=53 time=15.592 ms
64 bytes from 188.114.116.1: seq=1 ttl=53 time=18.736 ms
64 bytes from 188.114.116.1: seq=2 ttl=53 time=24.435 ms
64 bytes from 188.114.116.1: seq=3 ttl=53 time=18.367 ms
64 bytes from 188.114.116.1: seq=4 ttl=53 time=17.644 ms
64 bytes from 188.114.116.1: seq=5 ttl=53 time=17.342 ms
```
I still don't understand why manual IP causes the problem... Is this a bug or is it just me being thick?

**Comment by iteqla** [comment]
Also, why would it only work with pool.ntp.org and not with time.cloudfare.com? Is there any restriction?

```
[root@pi-bianco ~]# ping time.cloudfare.com
PING time.cloudfare.com (104.16.132.229): 56 data bytes
64 bytes from 104.16.132.229: seq=0 ttl=57 time=12.479 ms
64 bytes from 104.16.132.229: seq=1 ttl=57 time=15.909 ms
64 bytes from 104.16.132.229: seq=2 ttl=57 time=17.012 ms
64 bytes from 104.16.132.229: seq=3 ttl=57 time=14.902 ms
64 bytes from 104.16.132.229: seq=4 ttl=57 time=15.579 ms
```

**Comment by starbasessd** [comment]
Where did you set manual IP? 
In static_ip.conf, or somewhere else?

**Comment by starbasessd** [comment]
Did you enter the ip address as
192.168.1.16
or as 
192.168.1.16/24
?? You need to do it the second way...

**Comment by starbasessd** [comment]
time.cloudflare.com work for me as NTP.

**Comment by iteqla** [comment]
I did it on the settings here,
![2021-10-15 22_40_03-Window](https://user-images.githubusercontent.com/38462266/137556525-75ca08fc-0483-4b11-b51d-3c6c291d577f.jpg)

if I try to add /24 the address is not accepted. The subnet mask must be added explicitly as 255.255.255.0 on the following field.


![error](https://user-images.githubusercontent.com/38462266/137556785-f46bbb75-36db-4cdb-8209-6e09bb480cef.jpg)

**Comment by starbasessd** [comment]
Just tested 
cloudflare.com
as HTTP (with the reboot) and it's working

**Comment by iteqla** [comment]
I tried it too. Cloudfare works only with HTTP and only if the device gets an IP from the DHCP. 
If I set an IP address manually, no time server works at all... 

...it doesn't make any sense to me... could it be a bug?

**Comment by starbasessd** [comment]
Not a bug, AFAICT, I cannot reproduce it, similar setup. I also test with my network DHCP server issuing a static IP address, also no issue.
I would bet something is corrupt. If you have a spare thumb drive, and your Pi3B+ can boot to USB, dev20201026 supports booting to USB. You could test it that way.

**Comment by iteqla** [comment]
I guess you are right and the root cause was something that was corrupted somewhere.
I went the hard way: I formatted the SD card and started everything from scratch. Now the NTP works with both static and DHCP addresses... as long as I don't use cloudfare, which still have the same problem... but I guess this is a different type of problem, as pool.ntp.org is now no longer affected.
Thanks for your help and have a great weekend!
